packages/contracts/contracts/MarginlyPool.sol

```diff
@@ -785,35 +798,75 @@ contract MarginlyPool is IMarginlyPool {
     lastReinitTimestampSeconds = block.timestamp;
 
     FP96.FixedPoint memory secondsInYear = FP96.FixedPoint({inner: SECONDS_IN_YEAR_X96});
-    FP96.FixedPoint memory interestRate = FP96.FixedPoint({
-      inner: Math.mulDiv(params.interestRate, FP96.Q96, WHOLE_ONE)
-    });
+    FP96.FixedPoint memory interestRate = FP96.fromRatio(params.interestRate, WHOLE_ONE);
+    FP96.FixedPoint memory onePlusFee = FP96.fromRatio(params.fee, WHOLE_ONE).div(secondsInYear).add(FP96.one());
+
+    // FEE(dt) = (1 + fee)^dt
+    FP96.FixedPoint memory feeDt = FP96.powTaylor(onePlusFee, secondsPassed);
+
+    uint256 discountedBaseFee;
+    uint256 discountedQuoteFee;
+
     if (discountedBaseCollateral != 0) {
-      FP96.FixedPoint memory baseDebtCoeffMul = FP96.powTaylor(
-        interestRate.mul(FP96.FixedPoint({inner: systemLeverage.shortX96})).div(secondsInYear).add(FP96.one()),
-        secondsPassed
-      );
+      uint256 _discountedBaseDebt = discountedBaseDebt;
```

Comment:
You'd rather create variable like `uint256 realBaseDebtPrev = baseDebtCoeffPrev.mul(_discountedBaseDebt)` then